src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,616 @@
+                DacResidualUnit(output_dim, dilation=9),
+            ]
+        )
+
+    def forward(self, input):
+        for block in self.block: 
+            input = block(input)
+        return input
+
+
+class DacResidualVectorQuantize(nn.Module):
+    """
+    ResidualVectorQuantize block - Introduced in SoundStream: An end2end neural audio codec (https://arxiv.org/abs/2107.03312)
+    """
+    def __init__(
+        self,
+        config: DacConfig,
+    ):
+        super().__init__()
+
+        input_dim = config.latent_dim
+        n_codebooks = config.n_codebooks
+        codebook_size = config.codebook_size
+        codebook_dim = config.codebook_dim
+        quantizer_dropout = config.quantizer_dropout
+
+        if isinstance(codebook_dim, int):
+            codebook_dim = [codebook_dim for _ in range(n_codebooks)]
+
+        self.n_codebooks = n_codebooks
+        self.codebook_dim = codebook_dim
+        self.codebook_size = codebook_size
+
+        self.quantizers = nn.ModuleList(
+            [DacVectorQuantize(input_dim, codebook_size, codebook_dim[i]) for i in range(n_codebooks)]
+        )
+        self.quantizer_dropout = quantizer_dropout
+
+    def forward(self, input, n_quantizers: int = None):
+        """
+        Quantizes the input tensor using a fixed set of codebooks and returns corresponding codebook vectors.
+        Args:
+            input (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`):
+                Input tensor to be quantized.
+            n_quantizers (`int`, optional):
+                Number of quantizers to use. If specified and `self.quantizer_dropout` is True,
+                this argument is ignored during training, and a random number of quantizers is used.
+
+        Returns:
+            quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`, *optional*):
+                Quantized continuous representation of input.
+            codebook_indices (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`, *optional*):
+                Codebook indices for each codebook (quantized discrete representation of input).
+            projected_latents (`torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`, *optional*):
+                Projected latents (continuous representation of input before quantization).
+            commitment_loss (`torch.Tensor` of shape `(1)`, *optional*):
+                Commitment loss to train the encoder to predict vectors closer to codebook entries.
+            codebook_loss (`torch.Tensor` of shape `(1)`, *optional*):
+                Codebook loss to update the codebook.
+        """
+
+        quantized_representation = 0
+        residual = input
+        commitment_loss = 0
+        codebook_loss = 0
+
+        codebook_indices = []
+        projected_latents = []
+
+        if n_quantizers is None:
+            n_quantizers = self.n_codebooks
```

Comment:
(nit) usually we do single line `if`'s like this on one-line:
```suggestion
        n_quantizers = n_quantizers if n_quantizers is not None else self.n_codebooks
```